src/helpers/makeA11yQuery.ts

```diff
@@ -18,29 +18,33 @@ function makeAliases(aliases: Array<string>, query: Function) {
 }
 
 type QueryNames = {
-  getBy: Array<string>,
-  getAllBy: Array<string>,
-  queryBy: Array<string>,
-  queryAllBy: Array<string>,
-  findBy: Array<string>,
-  findAllBy: Array<string>,
+  getBy: Array<string>;
+  getAllBy: Array<string>;
+  queryBy: Array<string>;
+  queryAllBy: Array<string>;
+  findBy: Array<string>;
+  findAllBy: Array<string>;
 };
 
-const makeA11yQuery = <P: mixed, M: mixed>(
+const makeA11yQuery = <P extends unknown, M extends unknown>(
   name: string,
   queryNames: QueryNames,
   matcherFn: (prop: P, value: M) => boolean
-): ((instance: ReactTestInstance) => { ... }) => (
-  instance: ReactTestInstance
-) => {
+) => (instance: ReactTestInstance) => {
   const getBy = (matcher: M) => {
     try {
       return instance.find(
         (node) => isNodeValid(node) && matcherFn(node.props[name], matcher)
       );
     } catch (error) {
       throw new ErrorWithStack(
-        prepareErrorMessage(error, name, matcher),
+        prepareErrorMessage(
+          error instanceof Error
+            ? error
+            : new Error('Something wrong happened'),
```

Comment:
I'd suggest moving non-`Error` `error` handling inside the `preapareErrorMessage` so that it accepts `unknown` and then makes a check inside. As for other potential received values we could just call `error?.toString()` to get their printable value. @AugustinLF wdyt?